atomisticparsers/gromacs/parser.py

```diff
@@ -774,12 +774,51 @@ def get_mdp_file(self):
 
         return self.get_gromacs_file(self.mdp_ext)
 
-    def get_gromacs_file(self, ext):
+    def get_gromacs_file(self, ext, return_all=False):
         files = [d for d in self._gromacs_files if d.endswith(ext)]
 
         if len(files) == 0:
             return ''
 
+        if return_all:
+            all_files = {}
+            contain_basename, no_basename_match, counts = [], [], []
+            for f in files:
+                # We assume that the matching trajectory file has (or starts with)
+                # the same basename as the log file, e.g., out.log would correspond to
+                # out.tpr and out.trr and out.edr (or out_some.tpr ...).
+                # These files are therefore given first (second... ) priority in the dictionary.
+                if f.rsplit('.', 1)[0] == self._basename:
+                    all_files[0] = {0: os.path.join(self._maindir, f)}
+                elif f.rsplit('.', 1)[0].startswith(self._basename):
+                    contain_basename.append(os.path.join(self._maindir, f))
+                else:
+                    # if potential files are named differently, we guess that the ones with a unique name
+                    # would be the files we are more likely interested in, and sort the results accordingly
+                    count = 0
+                    for reff in self._gromacs_files:
+                        if f.rsplit('.', 1)[0] == reff.rsplit('.', 1)[0]:
+                            count += 1
+                    counts.append(count)
+                    no_basename_match.append(os.path.join(self._maindir, f))
+            # add the files starting with the same basename in the order they were encountered
+            if contain_basename:
```

Comment:
why not just create this dict as you go? ...also not quite sure yet why you are using a dict